app/controllers/main/routes.py

```diff
@@ -397,6 +397,51 @@ def deleteNote(username):
         flash("Failed to delete profile note", "danger")
     return "success"
 
+@main_bp.route('/<username>/editNote', methods=['POST'])
+def editProfileNote(username):
+    """
+    Edit an existing profile note.
+    """
+    profile_note_id = request.form.get('id')
+    new_text = (request.form.get('noteTextbox') or '').strip()
+    visibility = request.form.get('visibility')  # optional
+    bonner = request.form.get('bonner')          # optional
+
+    if not profile_note_id or not new_text:
+        return jsonify({"error": "Missing note id or content"}), 400
+
+    try:
+        pn = ProfileNote.get_by_id(profile_note_id)
```

Comment:
pn variable name seems very vague. Rename it to something more clear and elaborate